### ISSUE TITLE
Feature: Should warn about unquoted exports in /bin/sh

### ISSUE THREAD

**Issue by koalaman** [koalaman/shellcheck#951]
#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh
var="foo bar"
export foo=$var
```

#### Here's what shellcheck currently says:

Nothing

#### Here's what I wanted or expected to see:

The value for `export` must be quoted in POSIX sh

**Comment by mklement0** [comment]
Excellent idea; if the author of `/etc/profile.d/apps-bin-path.sh` that ships with Ubuntu 18.04 had been warned about this, https://github.com/PowerShell/PowerShell/issues/12020 wouldn't have happened.

**Comment by emorrp1** [comment]
This is also relevant to SC2046 for command substitution assignments, and interacts with SC2155, here's a few additional examples:

```
#!/bin/sh
var=$(cat) # does *not* need quoting
export var=$(cat) # SC2155
if var=$(cat); then
  true
fi
```
